game/gui/menu.py

```diff
@@ -0,0 +1,130 @@
+import pygame
+
+from constants import BLACK
+
+
+class Menu:
+    def __init__(self, display_manager):
+        self.display_manager = display_manager
+        self.mid_w, self.mid_h = self.display_manager.DISPLAY_W / 2, self.display_manager.DISPLAY_H / 2
+        self.run_display = True
+        self.cursor_rect = pygame.Rect(0, 0, 20, 20)
+        self.offset = - 100
+
+    def draw_cursor(self):
+        self.display_manager.draw_text('*', 15, self.cursor_rect.x, self.cursor_rect.y)
+
+    def blit_screen(self):
+        pygame.display.flip()
+        self.display_manager.reset_keys()
+
+
+class MainMenu(Menu):
+    def __init__(self, game):
+        Menu.__init__(self, game)
+        self.state = "Start"
+
+        # todo: remove magic numbers with font scaling or screen size
+        self.states = ("Start", "Options", "Help", "Credits", "Exit")
+        self.state_sz = len(self.states)
+        self.state_ind = 0
+        self.statex = tuple([self.mid_w for _ in range(self.state_sz)])
+        self.statey = tuple([self.mid_h + 30 + 20 * i for i in range(self.state_sz)])
+        self.cursor_rect.midtop = (self.statex[self.state_ind] + self.offset, self.statey[self.state_ind])
+
+    def display_menu(self):
+        self.run_display = True
+        while self.run_display:
+            self.display_manager.check_events()
+            self.check_input()
+            self.display_manager.screen.fill(BLACK)
+            self.display_manager.draw_text('Main Menu', 20, self.display_manager.DISPLAY_W / 2,
+                                           self.display_manager.DISPLAY_H / 2 - 20)
+            for i in range(self.state_sz):
+                self.display_manager.draw_text(self.states[i], 20, self.statex[i], self.statey[i])
+            self.draw_cursor()
+            self.blit_screen()
+
+    def move_cursor(self):
+        # scrolling up and down - keyboard controls
+        if self.display_manager.DOWN_KEY:
+            self.state_ind = (self.state_ind + 1) % self.state_sz
+            self.state = self.states[self.state_ind]
+            self.cursor_rect.midtop = (self.statex[self.state_ind] + self.offset, self.statey[self.state_ind])
+        elif self.display_manager.UP_KEY:
+            self.state_ind = (self.state_ind - 1) % self.state_sz
+            self.state = self.states[self.state_ind]
+            self.cursor_rect.midtop = (self.statex[self.state_ind] + self.offset, self.statey[self.state_ind])
+
+    def check_input(self):
+        # are some of the menues selected?
+        self.move_cursor()
+        if self.display_manager.START_KEY:
+            if self.state == 'Start':
+                self.display_manager.playing = True
+                self.display_manager.game.start_game()
```

Comment:
This is actually antipattern. Here you access game instance method (game instance is supposed to be responsible for holding game logic not GUI) right from GUI-related code through couple of layers of abstraction.

Reply:
I agree, it doesn't look that good here, but this only sets a boolean flag and starts the game thread: 
![slika](https://user-images.githubusercontent.com/81580576/235291910-404b3143-5e3c-48db-a4b6-25cd6478a7e0.png)
I will change this to be more clear and respect encapsulation.